From a879b83b9e11da8d0eb9cfde6e87274ed99c448e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?ChangZhuo=20Chen=20=28=E9=99=B3=E6=98=8C=E5=80=AC=29?=
 <czchen@debian.org>
Date: Wed, 15 Jun 2016 19:06:30 +0800
Subject: Fix FTBFS on mips, powerpc, s390x

 For some unknown reason, after right shifting, v is still greater than 256. It
 is possible cause by implementing right shift with rotation.
 .
 This patch just uses mask to fix v.
---
 tsa2d32.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tsa2d32.cpp b/tsa2d32.cpp
index d517b86..d0802a2 100644
--- a/tsa2d32.cpp
+++ b/tsa2d32.cpp
@@ -627,6 +627,8 @@ int main(int argc, char **argv)
       v = kk64 >> TSIN_HASH_SHIFT_64;
     }
 
+    v &= 0xff;
+
     if (v >= TSIN_HASH_N)
       p_err("error found %d", v);
 
